diff --git a/dist/index.cjs.js b/dist/index.cjs.js
index 900e3f3804257c4e61de76378f0c5c386b9acfe1..49e2458e614e25be176fdc6117b382f9c755c447 100644
--- a/dist/index.cjs.js
+++ b/dist/index.cjs.js
@@ -5,7 +5,6 @@ const context = require("./context-DS0wRSh7.cjs");
 const ReactRouterDOM = require("react-router-dom");
 const plugin = require("./plugin.cjs.js");
 const ReactDOM = require("react-dom");
-const ReactDOMClient = require("react-dom/client");
 function _interopNamespaceDefault(e2) {
   const n = Object.create(null, { [Symbol.toStringTag]: { value: "Module" } });
   if (e2) {
@@ -213,7 +212,7 @@ const RemoteAppWrapper = React.forwardRef(function(props, ref) {
       }
     );
   };
-  RemoteApp2["__APP_VERSION__"] = "0.8.10";
+  RemoteApp2["__APP_VERSION__"] = "0.8.9";
   return /* @__PURE__ */ React.createElement(RemoteApp2, null);
 });
 function withRouterData(WrappedComponent) {
@@ -296,14 +295,14 @@ function createLazyRemoteComponent(info) {
       exportName
     });
     try {
-      const m = await info.loader();
-      const moduleName = m && m[Symbol.for("mf_module_id")];
+      const m2 = await info.loader();
+      const moduleName = m2 && m2[Symbol.for("mf_module_id")];
       context.LoggerInstance.debug(
         `createRemoteComponent LazyComponent loadRemote info >>>`,
-        { name: moduleName, module: m, exportName }
+        { name: moduleName, module: m2, exportName }
       );
-      const exportFn = m[exportName];
-      if (exportName in m && typeof exportFn === "function") {
+      const exportFn = m2[exportName];
+      if (exportName in m2 && typeof exportFn === "function") {
         const RemoteAppComponent = React.forwardRef((props, ref) => {
           return /* @__PURE__ */ React.createElement(
             RemoteApp,
@@ -323,7 +322,7 @@ function createLazyRemoteComponent(info) {
       } else {
         context.LoggerInstance.debug(
           `createRemoteComponent LazyComponent module not found >>>`,
-          { name: moduleName, module: m, exportName }
+          { name: moduleName, module: m2, exportName }
         );
         throw Error(
           `Make sure that ${moduleName} has the correct export when export is ${String(
@@ -339,11 +338,41 @@ function createLazyRemoteComponent(info) {
 function createRemoteComponent(info) {
   return React.forwardRef(
     (props, ref) => {
-      const LazyComponent = createLazyRemoteComponent(info);
+      const LazyComponent = React.useMemo(() => createLazyRemoteComponent(info), [
+        info.loader,
+        info.loading,
+        info.fallback,
+        info.export
+        // Use stable parts of info as deps
+      ]);
       return /* @__PURE__ */ React.createElement(ErrorBoundary, { FallbackComponent: info.fallback }, /* @__PURE__ */ React.createElement(React.Suspense, { fallback: info.loading }, /* @__PURE__ */ React.createElement(LazyComponent, { ...props, ref })));
     }
   );
 }
+var client = {};
+var m = ReactDOM;
+if (process.env.NODE_ENV === "production") {
+  client.createRoot = m.createRoot;
+  client.hydrateRoot = m.hydrateRoot;
+} else {
+  var i = m.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
+  client.createRoot = function(c, o) {
+    i.usingClientEntryPoint = true;
+    try {
+      return m.createRoot(c, o);
+    } finally {
+      i.usingClientEntryPoint = false;
+    }
+  };
+  client.hydrateRoot = function(c, h, o) {
+    i.usingClientEntryPoint = true;
+    try {
+      return m.hydrateRoot(c, h, o);
+    } finally {
+      i.usingClientEntryPoint = false;
+    }
+  };
+}
 function createBridgeComponent(bridgeInfo) {
   return () => {
     const rootMap = /* @__PURE__ */ new Map();
@@ -397,7 +426,7 @@ function createBridgeComponent(bridgeInfo) {
               bridgeInfo == null ? void 0 : bridgeInfo.render(rootComponentWithErrorBoundary, dom)
             ).then((root) => rootMap.set(info.dom, root));
           } else {
-            const root = ReactDOMClient.createRoot(info.dom);
+            const root = client.createRoot(info.dom);
             root.render(rootComponentWithErrorBoundary);
             rootMap.set(info.dom, root);
           }
diff --git a/dist/index.es.js b/dist/index.es.js
index f3a8de5e3ecfaa5f4e2ded1d6e572423dc0292b5..c3cdf27b088ce1210b330aa29e169e0ac8ac598a 100644
--- a/dist/index.es.js
+++ b/dist/index.es.js
@@ -1,10 +1,9 @@
 import * as React from "react";
-import React__default, { createContext, Component, createElement, forwardRef, useRef, useEffect, useContext, useState } from "react";
+import React__default, { createContext, Component, createElement, forwardRef, useRef, useEffect, useContext, useState, useMemo } from "react";
 import { L as LoggerInstance, g as getRootDomDefaultClassName, p as pathJoin, a as atLeastReact18, R as RouterContext } from "./context-BgBVJ06L.js";
 import * as ReactRouterDOM from "react-router-dom";
 import { federationRuntime } from "./plugin.es.js";
 import ReactDOM from "react-dom";
-import ReactDOMClient from "react-dom/client";
 const ErrorBoundaryContext = createContext(null);
 const initialState = {
   didCatch: false,
@@ -194,7 +193,7 @@ const RemoteAppWrapper = forwardRef(function(props, ref) {
       }
     );
   };
-  RemoteApp2["__APP_VERSION__"] = "0.8.10";
+  RemoteApp2["__APP_VERSION__"] = "0.8.9";
   return /* @__PURE__ */ React__default.createElement(RemoteApp2, null);
 });
 function withRouterData(WrappedComponent) {
@@ -277,14 +276,14 @@ function createLazyRemoteComponent(info) {
       exportName
     });
     try {
-      const m = await info.loader();
-      const moduleName = m && m[Symbol.for("mf_module_id")];
+      const m2 = await info.loader();
+      const moduleName = m2 && m2[Symbol.for("mf_module_id")];
       LoggerInstance.debug(
         `createRemoteComponent LazyComponent loadRemote info >>>`,
-        { name: moduleName, module: m, exportName }
+        { name: moduleName, module: m2, exportName }
       );
-      const exportFn = m[exportName];
-      if (exportName in m && typeof exportFn === "function") {
+      const exportFn = m2[exportName];
+      if (exportName in m2 && typeof exportFn === "function") {
         const RemoteAppComponent = forwardRef((props, ref) => {
           return /* @__PURE__ */ React__default.createElement(
             RemoteApp,
@@ -304,7 +303,7 @@ function createLazyRemoteComponent(info) {
       } else {
         LoggerInstance.debug(
           `createRemoteComponent LazyComponent module not found >>>`,
-          { name: moduleName, module: m, exportName }
+          { name: moduleName, module: m2, exportName }
         );
         throw Error(
           `Make sure that ${moduleName} has the correct export when export is ${String(
@@ -320,11 +319,41 @@ function createLazyRemoteComponent(info) {
 function createRemoteComponent(info) {
   return forwardRef(
     (props, ref) => {
-      const LazyComponent = createLazyRemoteComponent(info);
+      const LazyComponent = useMemo(() => createLazyRemoteComponent(info), [
+        info.loader,
+        info.loading,
+        info.fallback,
+        info.export
+        // Use stable parts of info as deps
+      ]);
       return /* @__PURE__ */ React__default.createElement(ErrorBoundary, { FallbackComponent: info.fallback }, /* @__PURE__ */ React__default.createElement(React__default.Suspense, { fallback: info.loading }, /* @__PURE__ */ React__default.createElement(LazyComponent, { ...props, ref })));
     }
   );
 }
+var client = {};
+var m = ReactDOM;
+if (process.env.NODE_ENV === "production") {
+  client.createRoot = m.createRoot;
+  client.hydrateRoot = m.hydrateRoot;
+} else {
+  var i = m.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
+  client.createRoot = function(c, o) {
+    i.usingClientEntryPoint = true;
+    try {
+      return m.createRoot(c, o);
+    } finally {
+      i.usingClientEntryPoint = false;
+    }
+  };
+  client.hydrateRoot = function(c, h, o) {
+    i.usingClientEntryPoint = true;
+    try {
+      return m.hydrateRoot(c, h, o);
+    } finally {
+      i.usingClientEntryPoint = false;
+    }
+  };
+}
 function createBridgeComponent(bridgeInfo) {
   return () => {
     const rootMap = /* @__PURE__ */ new Map();
@@ -378,7 +407,7 @@ function createBridgeComponent(bridgeInfo) {
               bridgeInfo == null ? void 0 : bridgeInfo.render(rootComponentWithErrorBoundary, dom)
             ).then((root) => rootMap.set(info.dom, root));
           } else {
-            const root = ReactDOMClient.createRoot(info.dom);
+            const root = client.createRoot(info.dom);
             root.render(rootComponentWithErrorBoundary);
             rootMap.set(info.dom, root);
           }
