diff --git a/src/plugins/plugin.js b/src/plugins/plugin.js
index 6850cb0144d87efdbcb10ac4274532adcc16be6c..c5cbbbb918639ba96f076dfa8d8a88949051fe88 100644
--- a/src/plugins/plugin.js
+++ b/src/plugins/plugin.js
@@ -80,20 +80,20 @@ function buildTarget(buildTargetName, projectRoot, buildDirectory, assetsBuildDi
                 : ['default', '^default']),
         ],
         outputs: [serverBuildOutputPath, assetsBuildOutputPath],
-        command: 'remix build',
+        command: 'remix vite:build',
         options: { cwd: projectRoot },
     };
 }
 function devTarget(serverBuildPath, projectRoot) {
     return {
-        command: 'remix dev --manual',
+        command: 'remix vite:dev',
         options: { cwd: projectRoot },
     };
 }
 function startTarget(projectRoot, serverBuildPath, buildTargetName) {
     return {
         dependsOn: [buildTargetName],
-        command: `remix-serve ${serverBuildPath}`,
+        command: `vite preview ${serverBuildPath}`,
         options: {
             cwd: projectRoot,
         },
