diff --git a/pluralize.js b/pluralize.js
index b8f970adc84a99de69f316ac05358130dc7aa0ce..42484a163636ac56970d09758bb7261550fb1832 100644
--- a/pluralize.js
+++ b/pluralize.js
@@ -1,20 +1,23 @@
 /* global define */
 
-(function (root, pluralize) {
+(function(root, pluralize) {
   /* istanbul ignore else */
   if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
-    // Node.
+    // Node (CommonJS)
     module.exports = pluralize();
   } else if (typeof define === 'function' && define.amd) {
-    // AMD, registers as an anonymous module.
-    define(function () {
+    // AMD
+    define(function() {
       return pluralize();
     });
+  } else if (typeof import.meta !== 'undefined') {
+    // ESM
+    return pluralize();
   } else {
-    // Browser global.
+    // Browser global
     root.pluralize = pluralize();
   }
-})(this, function () {
+})(this, function() {
   // Rule storage - pluralize and singularize need to be run sequentially,
   // while other rules can be optimized using an object for instant lookups.
   var pluralRules = [];
