diff --git a/dist/index.esm.mjs b/dist/index.esm.mjs
index 0469f0254868463b7ac281c5cbd409275dee6728..0bcbd606424f99b8124aea0129d2664610c02659 100644
--- a/dist/index.esm.mjs
+++ b/dist/index.esm.mjs
@@ -2680,5 +2680,5 @@ function useForm(props = {}) {
     }), [formState, control]);
 }
 
-export { Controller, Form, FormProvider, appendErrors, get, set, useController, useFieldArray, useForm, useFormContext, useFormState, useWatch };
+export { Controller, Form, FormProvider, appendErrors, get, set, useController, useFieldArray, useForm, useFormContext, useFormState, useWatch, HookFormContext };
 //# sourceMappingURL=index.esm.mjs.map
diff --git a/dist/useFormContext.d.ts b/dist/useFormContext.d.ts
index a418b7a0ed9bcc4f4280d531238d76ebf1b17b99..46e7935489e38f898c199b2a3bc452b781726177 100644
--- a/dist/useFormContext.d.ts
+++ b/dist/useFormContext.d.ts
@@ -31,6 +31,8 @@ import { FieldValues, FormProviderProps, UseFormReturn } from './types';
  * ```
  */
 export declare const useFormContext: <TFieldValues extends FieldValues, TContext = any, TransformedValues extends FieldValues | undefined = undefined>() => UseFormReturn<TFieldValues, TContext, TransformedValues>;
+
+export declare const HookFormContext: React.Context<UseFormReturn | null>;
 /**
  * A provider component that propagates the `useForm` methods to all children components via [React Context](https://reactjs.org/docs/context.html) API. To be used with {@link useFormContext}.
  *
