diff --git a/node_modules/tus-js-client/lib.es5/upload.js b/node_modules/tus-js-client/lib.es5/upload.js
index fe647f2..a2fbf56 100644
--- a/node_modules/tus-js-client/lib.es5/upload.js
+++ b/node_modules/tus-js-client/lib.es5/upload.js
@@ -724,7 +724,6 @@ var BaseUpload = /*#__PURE__*/function () {
               parallelUploads: 1,
               // Reset this option as we are not doing a parallel upload.
               parallelUploadBoundaries: null,
-              metadata: {},
               // Add the header to indicate the this is a partial upload.
               headers: _objectSpread(_objectSpread({}, _this3.options.headers), {}, {
                 'Upload-Concat': 'partial'
diff --git a/node_modules/tus-js-client/lib.esm/upload.js b/node_modules/tus-js-client/lib.esm/upload.js
index 02fde5b..f9e3972 100644
--- a/node_modules/tus-js-client/lib.esm/upload.js
+++ b/node_modules/tus-js-client/lib.esm/upload.js
@@ -272,7 +272,6 @@ var BaseUpload = /*#__PURE__*/function () {
               parallelUploads: 1,
               // Reset this option as we are not doing a parallel upload.
               parallelUploadBoundaries: null,
-              metadata: {},
               // Add the header to indicate the this is a partial upload.
               headers: _objectSpread(_objectSpread({}, _this3.options.headers), {}, {
                 'Upload-Concat': 'partial'
diff --git a/node_modules/tus-js-client/lib/upload.js b/node_modules/tus-js-client/lib/upload.js
index 61289da..0cdb559 100644
--- a/node_modules/tus-js-client/lib/upload.js
+++ b/node_modules/tus-js-client/lib/upload.js
@@ -322,7 +322,6 @@ class BaseUpload {
               parallelUploads: 1,
               // Reset this option as we are not doing a parallel upload.
               parallelUploadBoundaries: null,
-              metadata: {},
               // Add the header to indicate the this is a partial upload.
               headers: {
                 ...this.options.headers,
